### PR TITLE
Add documentation for comparing json and ordered_json

### LINKED ISSUE #3443: ordered_json vs json types comparison
nlohman compares jsons as strings so comparing 2 equal `nlohmann::ordered_json` objects could produce wrong results: 

```
using json = nlohmann::ordered_json; // ordered to preserve the insertion order for print
using ujson = nlohmann::json; // unordered json for testing comparison

    json j1 = { {"version", 1}, {"type", "mytype"} };
    json j2 = { {"type", "mytype"}, {"version", 1} };
    ujson j3 = ujson(j1);
    ujson j4 = ujson(j2);

    // nlohman compares jsons as strings, so...
    BOOST_CHECK_NE(j1, j2);     // these are not equal
    BOOST_CHECK_EQUAL(j3, j4);  // these are equal
```

So I would change `operator==` to this:
```
inline bool operator==(const nlohmann::ordered_json& lhs, const nlohmann::ordered_json& rhs) 
{
    return nlohmann::json(lhs) == nlohmann::json(rhs);
}
```

and add operator to compare `json` and `nlohmann::ordered_json`:

```
inline bool operator==(const nlohmann::json& lhs, const nlohmann::ordered_json& rhs) 
{
    return lhs == nlohmann::json(rhs);
}
```

and vice versa:

```
inline bool operator==(const nlohmann::ordered_json& lhs, const nlohmann::json& rhs) 
{
    return nlohmann::json(lhs) == rhs;
}
```
What do you think?

### Compiler and operating system

gcc 9.4.0, Ubuntu 20.04.4 LTS

### Library version

3.10.5

### Validation

- [ ] The bug also occurs if the latest version from the [`develop`](https://github.com/nlohmann/json/tree/develop) branch is used.
- [ ] I can successfully [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests).

### PR DESCRIPTION
**Pull request by nlohmann**
Closes  #3443

### CODE REVIEW

File: docs/mkdocs/docs/api/basic_json/operator_eq.md [hunk 1]
```diff
@@ -99,6 +99,39 @@ Linear.
...
+    and
+
+    ```json
+    {
+       "type": "integer",
+       "version": 1
+    }
+    ```
+
+    depends on whether [`nlohmann::json`](../json.md) order [`nlohmann::ordered_json`](../ordered_json.md) is used:
```
Comment (falbrechtskirchinger) [inline docs/mkdocs/docs/api/basic_json/operator_eq.md:123]: "or" oder "order"? :-)

```suggestion
    depends on whether [`nlohmann::json`](../json.md) or [`nlohmann::ordered_json`](../ordered_json.md) is used:
```

File: docs/mkdocs/docs/api/basic_json/operator_eq.md [hunk 2]
```diff
@@ -99,6 +99,39 @@ Linear.
     }
     ```
 
+!!! note "Comparing different `basic_json` specializations"
+
+    Comparing different `basic_json` specializations can have surprising results. For instance, comparing the JSON
+    objects
```
Comment (falbrechtskirchinger) [inline docs/mkdocs/docs/api/basic_json/operator_eq.md:105]:
```suggestion
    Comparing different `basic_json` specializations can have surprising effects. For instance, the result of comparing
    the JSON objects
```

File: docs/mkdocs/docs/api/basic_json/operator_eq.md (hunk 1 above)
Comment (falbrechtskirchinger) [inline docs/mkdocs/docs/api/basic_json/operator_eq.md:123]: "or" oder "order"? :-)

```suggestion
    depends on whether [`nlohmann::json`](../json.md) or [`nlohmann::ordered_json`](../ordered_json.md) is used:
```

**Review by falbrechtskirchinger — CHANGES_REQUESTED** [review]

**Comment by akontsevich** [comment]
I would extend the doc. Still not clear how to compare different JSON types. I converted one to another first.

**Pull request merged**